Gems/MiniAudio/Code/CMakeLists.txt

```diff
@@ -0,0 +1,217 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# Currently we are in the Code folder: ${CMAKE_CURRENT_LIST_DIR}
+# Get the platform specific folder ${pal_dir} for the current folder: ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME}
+# Note: o3de_pal_dir will take care of the details for us, as this may be a restricted platform
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/MiniAudio/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/MiniAudio/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
+
+# Now that we have the platform abstraction layer (PAL) folder for this folder, thats where we will find the
+# traits for this platform. Traits for a platform are defines for things like whether or not something in this gem
+# is supported by this platform.
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+# The MiniAudio.API target declares the common interface that users of this gem should depend on in their targets
+ly_add_target(
+    NAME MiniAudio.API INTERFACE
```

Comment:
The MiniAudio targets can be substituted with `${gem_name}` to make it easier to rename later. 
This needs to be combined with the `o3de_gem_setup` call from the previous commen.t